File: Day20.cpp
class Solution {
public:
    string intToRoman(int num) {
        string rom[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        int val[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string ans;
        int i = 0;
        while(num!=0){
            while(num >= val[i]){
                num -= val[i];
                ans += rom[i];
            }
            i++;
        }
        return ans;
    }
};

File: Day20.java
class Solution {
    public String intToRoman(int num) {
        HashMap<Integer,String> map = new LinkedHashMap<>();
        map.put(1000,"M");
        map.put(900,"CM");
        map.put(500,"D");
        map.put(400,"CD");
        map.put(100,"C");
        map.put(90,"XC");
        map.put(50,"L");
        map.put(40,"XL");
        map.put(10,"X");
        map.put(9,"IX");
        map.put(5,"V");
        map.put(4,"IV");
        map.put(1,"I");
        
        StringBuilder roman = new StringBuilder();
        for(int ele :map.keySet()){
            while(num>=ele){
                num -=ele;
                roman.append(map.get(ele));
            }
        }
        return roman.toString();
    }
}
